pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1032,19 +1034,20 @@ cdef class _Timestamp(ABCTimestamp):
         stamp = self._repr_base
         zone = None
 
-        try:
-            stamp += self.strftime("%z")
-        except ValueError:
-            year2000 = self.replace(year=2000)
-            stamp += year2000.strftime("%z")
+        if self.tzinfo is not None:
+            try:
+                stamp += self.strftime("%z")
+            except ValueError:
+                year2000 = self.replace(year=2000)
+                stamp += year2000.strftime("%z")
 
-        if self.tzinfo:
-            zone = get_timezone(self.tzinfo)
-        try:
-            stamp += zone.strftime(" %%Z")
-        except AttributeError:
-            # e.g. tzlocal has no `strftime`
-            pass
+            if self.tzinfo:
+                zone = get_timezone(self.tzinfo)
```

Comment:
this could now be simplified, right? you've already checked `self.tzinfo` above
```suggestion
            zone = get_timezone(self.tzinfo)
```

Reply:
IIRC there was an array_strftime-with-non-nano issue a few weeks ago, might this help at least some cases?